tbump/hooks.py

```diff
@@ -44,11 +45,28 @@ def __init__(self, working_path: Path) -> None:
     def add_hook(self, hook: Hook) -> None:
         self.hooks.append(hook)
 
-    def run(self, new_version: str) -> None:
-        for i, hook in enumerate(self.hooks):
-            hook.cmd = hook.cmd.format(new_version=new_version)
-            ui.info_count(i, len(self.hooks), ui.bold, hook.name)
+    def run_hooks_before_push(self, new_version: str) -> None:
+        self._run(new_version, after_push=False)
+
+    def run_hooks_after_push(self, new_version: str) -> None:
+        self._run(new_version, after_push=True)
+
+    def _run(self, new_version: str, *, after_push: bool = False) -> None:
+        matching_hooks = [hook for hook in self.hooks if hook.after_push == after_push]
+        if after_push:
+            desc = "after push"
+        else:
+            desc = "before push"
+        if matching_hooks:
+            ui.info_2("Running hooks", desc)
+        for i, hook in enumerate(matching_hooks):
+            try:
+                hook.cmd = hook.cmd.format(new_version=new_version)
+            except KeyError:
+                pass
```

Comment:
Maybe comment what throws KeyError and why you should ignore it. Aren't you hiding an error in case the user made a typo in `{new_version}`?